pygame_gui/elements/ui_screen_space_health_bar.py

```diff
@@ -34,209 +36,52 @@ def __init__(self,
+                         percent_method=self.health_percent,
+                         container=container,
+                         parent_element=parent_element,
+                         object_id=object_id,
                          anchors=anchors,
                          visible=visible)
 
-        self._create_valid_ids(container=container,
-                               parent_element=parent_element,
-                               object_id=object_id,
-                               element_id='screen_space_health_bar')
-
-        self.current_health = 50
-        self.health_capacity = 100
-        self.health_percentage = self.current_health / self.health_capacity
-
-        self.font = None
-        self.border_width = None
-        self.shadow_width = None
-        self.border_colour = None
-        self.bar_unfilled_colour = None
-        self.bar_filled_colour = None
-        self.text_shadow_colour = None
-        self.text_colour = None
-        self.text_horiz_alignment = 'center'
-        self.text_vert_alignment = 'center'
-        self.text_horiz_alignment_padding = 1
-        self.text_vert_alignment_padding = 1
-
-        self.border_rect = None
-        self.capacity_width = None
-        self.capacity_height = None
-        self.capacity_rect = None
-        self.current_health_rect = None
-
-        self.drawable_shape = None
-        self.shape = 'rectangle'
-        self.shape_corner_radius = None
-
-        if sprite_to_monitor is not None:
+    @property
+    def current_health(self):
+        if not self.sprite:
+            return 50
+        return self.sprite.current_health
+
+    @property
+    def health_capacity(self):
+        if not self.sprite:
+            return 100
+        return self.sprite.health_capacity
+
+    def health_percent(self):
+        return self.current_health / self.health_capacity
+
+    @property
+    def health_percentage(self):
+        # Now that we subclass UIStatusBar, this is here for backward compatibility.
+        return self.health_percent()
+
+    def set_sprite_to_monitor(self, sprite_to_monitor: pygame.sprite.Sprite):
+        if sprite_to_monitor:
             if not hasattr(sprite_to_monitor, 'health_capacity'):
                 raise AttributeError('Sprite does not have health_capacity attribute')
             if not hasattr(sprite_to_monitor, 'current_health'):
                 raise AttributeError('Sprite does not have current_health attribute')
-            self.sprite_to_monitor = sprite_to_monitor
-        else:
-            self.sprite_to_monitor = None
-        self.set_image(None)
-        self.background_text = None
-        self.foreground_text = None
+        self.sprite = sprite_to_monitor
 
-        self.rebuild_from_changed_theme_data()
+    def status_text(self):
+        """ Subclass and override this method to change what text is displayed, or to suppress the text. """
+        return f"{self.current_health:0.1f}/{self.health_capacity:0.1f}"
```

Comment:
This is a change from the previous display string which was two integers. I suspect integers for health is a more common design than fractional health points - but I suppose it could also be an optional theming parameter to set the precision of the display string? Would also have to switch away from f strings if precision is going to be a themed thing as f-strings don't support deferred evaluation.

Either way I think I would default this to integer level precision (i.e. '50/100' ).

Reply:
I'm pretty sure I copy/pasted what was there for this (and I created this separate method explicitly because the float values being displayed annoyed me), but in any case I'm happy to use integers.

Reply:
Can you figure out why UIProgressBar isn't showing up in the docs TOC? I couldn't figure out anything I was doing differently than for other classes?